### ISSUE TITLE
Run time config objects should be disallowed in build steps completely

### ISSUE THREAD

**Issue by dmlloyd** [quarkusio/quarkus#5514]
**Describe the bug**
Users are using run time configuration object values in build steps, expecting the effects to happen at run time.  See https://github.com/quarkusio/quarkus/issues/5457#issuecomment-554270894 for some examples.

Usage of fields of run time configuration objects during build steps should be marked as an error.  However this is impossible unfortunately.  So the next best thing would be to ban run time configuration objects completely from build steps.  Their only use is to be passed in to recorder instances; the recorders should be able to directly inject the run time config instead.  Then, any usage of a run time configuration object in a build step can be marked as an error.

**Comment by geoand** [comment]
This is now possible due to https://github.com/quarkusio/quarkus/pull/18983.

However making this change would be a breaking change.

**Comment by dmlloyd** [comment]
We could start with a warning... or better yet, make it an error with a switch to change the error to a warning, to be sure we've gotten the attention of the user.

**Comment by geoand** [comment]
Yeah, the only thing I am worried about is extensions in the Quarkiverse that are still using the old way of doing things that might not be maintained very actively.

Maybe just start with a warning? WDYT @gsmet ?
